spec/formatting.md

```diff
@@ -106,16 +112,44 @@ In a _pattern_, the resolved value of an _expression_ or _markup_ is used in its
 The form that resolved values take is implementation-dependent,
 and different implementations MAY choose to perform different levels of resolution.
 
-> For example, the resolved value of the _expression_ `{|0.40| :number style=percent}`
-> could be an object such as
+> While this specification does not require it,
+> a _resolved value_ could be implemented by requiring each function implementation to
+> return a value matching the following interface:
 >
+> ```ts
+> interface MessageValue {
+>   formatToString(): string
+>   formatToX(): X // where X is an implementation-defined type
+>   getValue(): unknown
+>   resolvedOptions(): { [key: string]: unknown }
+>   selectKeys(keys: string[]): string[]
+> }
 > ```
-> { value: Number('0.40'),
->   formatter: NumberFormat(locale, { style: 'percent' }) }
-> ```
 >
-> Alternatively, it could be an instance of an ICU4J `FormattedNumber`,
-> or some other locally appropriate value.
+> With this approach:
+> - An _expression_ could be used as a _placeholder_ if
+>   calling the `formatToString()` or `formatToX()` method of its _resolved value_
+>   did not throw an error.
+> - An _expression_ could be used as a _selector_ _expression_ if
+>   calling the `selectKeys(keys)` method of its _resolved value_
+>   did not throw an error.
+> - An _expression_ could be used as an _operand_ or _option_ value if
```

Comment:
Expression here is wrong? You can't put an _expression_ as an operand or option value, but you can reference the annotated value.

Would this actually be "the resolved value of an expression could be used..."?

Reply:
```suggestion
> - Using a _variable reference_, the resolved value of an _expression_
>   could be used as an _operand_ or _option_ value if
```